packages/map-template/src/components/Wayfinding/Wayfinding.jsx

```diff
@@ -404,16 +440,19 @@ function Wayfinding({ onStartDirections, onBack, directionsToLocation, direction
                             cleared={() => onSearchCleared(searchFieldIdentifiers.TO)}
                         />
                     </label>
-                    {userPosition && originLocation?.id !== 'USER_POSITION' && <p className="wayfinding__use-current-position">
-                        <button onClick={() => setMyPositionAsOrigin()}>Use My Position</button>
-                    </p>}
+
                 </div>
             </div>
             {!hasFoundRoute && <p className="wayfinding__error">No route found</p>}
             {!hasSearchResults && <p className="wayfinding__error">Nothing was found</p>}
+            {userPosition && !myPositionSelected && showMyPosition && <button className="wayfinding__use-current-position" onClick={() => selectMyPosition()}>
```

Comment:
```suggestion
            {userPosition && !myPositionSelected && showMyPosition && <button type="button" className="wayfinding__use-current-position" onClick={() => selectMyPosition()}>
```

just in case the button is suddenly put inside a form element 😄